### ISSUE TITLE
Add facet by controls

### ISSUE THREAD

**Issue by metasoarous** [matsengrp/olmsted#34]
This should default to nothing, as it is now, but also let you facet by data, subject, timepoint, subject/timepoint (repertoire), and any of the nominal variable which might be shared between them (see #33).

The goal here is that this can let us compare the distribution of x/y/color/shape patterns between whatever we're faceting by. So this could for example be used to see how some set of variables changes over time in relation to each other in a subject for whom we have multiple timepoints. Or a 

This is extracted from #5.

**Comment by eharkins** [comment]
Working on this has raised some questions. @metasoarous, let me know what you think.

Do we want to facet 
What sort of behavior do we want for selection results.

**Comment by metasoarous** [comment]
I was definitely thinking about B, but A would be super useful as well, particularly once #31 is in place and we have more variables to actually compare. Let's add a separate issue for that.

As for selections, I was initially thinking that selection would just be from within a single pane of the facet, and suspect that may be the most expected behavior. But I realize that is also a bit more complicated, and playing with the example you linked (note the shorter markdown url edit I made), the solution of just selecting across panes is fine for now. Maybe this could be another issue as well?

**Comment by eharkins** [comment]
Been working on this in https://github.com/matsengrp/olmsted/tree/34-facet, I believe if we want to be able to change the field by which we facet, we will need to have that be at the react level. This works in my mind since the facet-able spec will definitely need to be separate from our original spec; so the user can select in a react component dropdown a field by which to facet, and if the value of the dropdown is undefined we will render the normal plot, while if it is defined, we will pass that to the function that creates the faceted spec and render that. Thoughts on this @metasoarous?

**Comment by metasoarous** [comment]
Why does this have to be done at the react level? Does this have to do with where you can use signals, or is there something specific about the way facets work that force our hands here? It would be nice if we had a clear picture of where we can do these sorts of signal things and where we can't.

Fine to go this way if that's what is required. Just want to make sure.

**Comment by eharkins** [comment]
>Does this have to do with where you can use signals

That's what I suspected and I still think this could be true.

>It would be nice if we had a clear picture of where we can do these sorts of signal things and where we can't.

I agree; at first I took the docs' word for it when they have things like this:
![screen shot 2018-10-26 at 9 33 26 am](https://user-images.githubusercontent.com/12140437/47579899-39f01f00-d902-11e8-9dd8-ed063b8f8759.png)
But that particular example actually allows you to specify `{"signal": "signalNameX" }` as do many other fields which they require a specific type for, which is sort of misleading. That being said, it might have to do less with where you can specify a signal instead of a value, and more with which fields will actually update when you change that signal via a bound selection like a dropdown. E.g. I got the faceted field to a point where it was a signal reference everywhere and not hardcoded anywhere, and then I tried updating that signal with a bound selection (dropdown) and it didn't work. But there could be some other reason for that.

**Comment by eharkins** [comment]
Merged https://github.com/matsengrp/olmsted/tree/34-facet. We are able to just use the faceted spec and if we want to just have a single plot, we use the default facet field called "none", which we explicitly assign as a dummy field in a transform of our data. Because all data has the same value for the "none" field, it all shows up in one plot and we can hide the facet sub-plot labels. This of course means we couldn't have a field called "none" in our data since it gets overwritten, but my view on why that is ok is this: 

Anytime we use a [formula transform](https://vega.github.io/vega/docs/transforms/formula/) and add a field to our data, we are assuming the data will not have said field already. This is ok, so long as we are pretty sure for the sake of the viz (since we are doing this only in vega not changing the original data) that we are not writing over important data in a field with that same name, we can use that namespace for our derived data or in the case above, our dummy field.

A separate issue with the merged facet spec is #105. But that has to do with the vega errors that happen specific to faceting, which are outlined in that issue and the vega issue it references
